Jenkinsfile

```diff
@@ -0,0 +1,121 @@
+def user_id
+def group_id
+node {
+    user_id = sh(returnStdout: true, script: 'id -u').trim()
+    group_id = sh(returnStdout: true, script: 'id -g').trim()
+}
+
+def setupScalaBuildTool() {
+    sh "./ops/jenkins/pre-build.sh"
+}
+
+pipeline {
+    agent {
+        dockerfile {
+            dir '.ops/build-docker'
```

Comment:
This is referencing items in the main stardog repo, but this is it's own repo.

I don't believe the jenkinsfile can depend on items from another repo like this since the location of this file is tied with the repo.

In cases where the stardog pipeline uses other repos, all the base support is in the stardog repo and then we clone the needed repos and run from there. That's one option we can do here (e.g. keep this jenkinsfile and whatnot in the stardog repo like we do for starbench but then have this pipeline clone the jedai repo and run whatever tests/scripts you want).

It's also fine to have this as it's own thing in this repo, but then you will need to add your own stripped down dockerfile that builds everything you need for this stuff.

Reply:
> This is referencing items in the main stardog repo, but this is it's own repo.

My bad. I should have seen that. Missed it as haven't run it through Jenkins yet.

This repo does not depend on Stardog so I believe we may not need all the stuff that we have under Dockerfile of Stardog repo. Since starbench depends on Stardog so it would make sense to have base stuff in Stardog itself. 

However, in our case (jedai-spark), I think we can create a trimmed down version of Dockerfile. Thoughts?

Reply:
Yep, that all makes sense to me. A slimmed down version of what you need sounds good. I think going forward we'll be basing a lot of our CI images on the latest LTS ubuntu docker images -- official decision is still pending -- that are pretty slim (then you can install what you need). So basing it on something like the `ubuntu:22.04` docker image (assuming that works with what you need) will likely be in line with stuff we do in other places of CI going forward just FYI.

Reply:
I have added the slimmed down Dockerfile (with ubuntu:22.04) and tested locally on my system. It is installing the necessary libraries. I could not test the code on Jenkins as I couldn't see "Build with parameters" option. I tried to make the build parameterized but did not see that on Jenkins either.

Reply:
Looks like it failed to compile as `JAVA_VERSION` is the `choice` parameter and there is no choice provided while defining it. Making necessary changes.

Reply:
@pdmars I have made the necessary changes. Jenkins build is running successfully and uploading artifacts on jfrog. I verified by adding its dependency in the separate project.

I have also moved `jenkins` folder from `.ops` to `ops`. I found this folder structure in `stardog` repository so did this change.

Couple of queries:
I think we do not need `stardog-testing/nightly-develop-jedai-snapshot/binaries/` directory when all the required artifacts are uploaded under `MAVEN` folder. I can remove the code associated with this.

Secondly, in `Dockerfile`, I found `openjdk` installed under `/usr/lib/jvm/java-11-openjdk-arm64` while in case of Jenkins it was `/usr/lib/jvm/java-11-openjdk-amd64`. Couldn't figure out.

Reply:
Yeah no need to do anything with `nightly-develop-jedai-snapshot/binaries/` if we are not using it. Just uploading the maven artifacts, if that's all you need, is fine.

Reply:
Removed that code.